examples/Prioritized_DQN/atari/agent.py

```diff
@@ -0,0 +1,112 @@
+    def __init__(self, algorithm, act_dim, update_freq):
+        super(AtariAgent, self).__init__(algorithm)
+        assert isinstance(act_dim, int)
+        self.act_dim = act_dim
+        self.exploration = 1.0
+        self.global_step = 0
+        self.update_target_steps = 10000 // 4
+        self.update_freq = update_freq
+
+    def build_program(self):
+        self.pred_program = fluid.Program()
+        self.learn_program = fluid.Program()
+
+        with fluid.program_guard(self.pred_program):
+            obs = layers.data(
+                name='obs',
+                shape=[CONTEXT_LEN, IMAGE_SIZE[0], IMAGE_SIZE[1]],
+                dtype='float32')
+            self.value = self.alg.predict(obs)
+
+        with fluid.program_guard(self.learn_program):
+            obs = layers.data(
+                name='obs',
+                shape=[CONTEXT_LEN, IMAGE_SIZE[0], IMAGE_SIZE[1]],
+                dtype='float32')
+            action = layers.data(name='act', shape=[1], dtype='int32')
+            reward = layers.data(name='reward', shape=[], dtype='float32')
+            next_obs = layers.data(
+                name='next_obs',
+                shape=[CONTEXT_LEN, IMAGE_SIZE[0], IMAGE_SIZE[1]],
+                dtype='float32')
+            lr = layers.data(
+                name='lr', shape=[1], dtype='float32', append_batch_size=False)
+            terminal = layers.data(name='terminal', shape=[], dtype='bool')
+            ISweight = layers.data(name='ISweight', shape=[1], dtype='float32')
+            self.cost, self.delta = self.alg.learn(
+                obs, action, reward, next_obs, terminal, lr, ISweight)
+
+    def sample(self, obs, decay_exploration=True):
+        sample = np.random.random()
+        if sample < self.exploration:
+            act = np.random.randint(self.act_dim)
+        else:
+            if np.random.random() < 0.01:
+                act = np.random.randint(self.act_dim)
+            else:
+                obs = np.expand_dims(obs, axis=0)
+                pred_Q = self.fluid_executor.run(
+                    self.pred_program,
+                    feed={'obs': obs.astype('float32')},
+                    fetch_list=[self.value])[0]
+                pred_Q = np.squeeze(pred_Q, axis=0)
+                act = np.argmax(pred_Q)
+        if decay_exploration:
+            self.exploration = max(0.1, self.exploration - 1e-6)
+        return act
+
+    def predict(self, obs):
+        obs = np.expand_dims(obs, axis=0)
+        pred_Q = self.fluid_executor.run(
+            self.pred_program,
+            feed={'obs': obs.astype('float32')},
+            fetch_list=[self.value])[0]
+        pred_Q = np.squeeze(pred_Q, axis=0)
+        act = np.argmax(pred_Q)
+        return act
+
+    def learn(self, obs, act, reward, next_obs, terminal, ISweight):
+        if self.global_step % self.update_target_steps == 0:
+            self.alg.sync_target()
+        self.global_step += 1
+
+        lr = 0.00025 / 4
```

Comment:
When using a stable learning rate, we should put the definition of learning rate in algorithm. Please refer to the lr in PolicyGradient.
https://github.com/PaddlePaddle/PARL/blob/a50793e43acdbf5c27d5fd96bcec8b76dc777c0d/parl/algorithms/fluid/policy_gradient.py#L26